rules/alerting.go

```diff
@@ -404,16 +415,29 @@ func (r *AlertingRule) Eval(ctx context.Context, ts time.Time, query QueryFunc,
 	// Check if any pending alerts should be removed or fire now. Write out alert timeseries.
 	for fp, a := range r.active {
 		if _, ok := resultFPs[fp]; !ok {
+			var keepFiring bool
```

Comment:
I would find it useful to have a comment hereabouts saying something like "no results for this fingerprint; the alert is no longer firing".